guides/source/active_model_basics.md

```diff
@@ -122,68 +400,237 @@ class Person
+passing the object that the callback is being called on.
+
+```ruby
+  class MyModel
+    extend ActiveModel::Callbacks
+    define_model_callbacks :create
+
+    before_create AnotherClass
+  end
+
+  class AnotherClass
+    def self.before_create(obj)
+      # obj is the MyModel instance that the callback is being called on
+    end
+  end
+```
+
+NOTE: `method_name` passed to `define_model_callbacks` must not end with `!`,
+`?` or `=`. In addition, defining the same callback multiple times will
+overwrite previous callback definitions.
+
+Like the Active Record methods, the callback chain is aborted as soon as one of
+the methods throws `:abort`.
+
 ### Conversion
 
-If a class defines `persisted?` and `id` methods, then you can include the
-`ActiveModel::Conversion` module in that class, and call the Rails conversion
-methods on objects of that class.
+`ActiveModel::Conversion` is a collection of methods that allow you to convert
+your object to different forms for different purposes. A common use case is to
+convert your object to a string or an integer to build URLs, form fields, and
+more.
+
+The `ActiveModel::Conversion` module adds the following methods: `to_model`,
+`to_key`, `to_param`, and `to_partial_path` to classes.
+
+The return values of the methods depend on whether `persisted?` is defined and
+if an `id` is provided. The `persisted?` method should return true if the object
+has been saved to the database or store, otherwise, it should return `false`.
+The `id` should reference the id of the object or nil if the object is not
+saved.
 
 ```ruby
 class Person
   include ActiveModel::Conversion
 
-  def persisted?
-    false
+  def initialize(id)
+    @id = id
   end
 
-  def id
-    nil
+  def persisted?
+    true
   end
 end
 ```
 
+#### to_model
+
+The `to_model` method returns the object itself.
+
 ```irb
-irb> person = Person.new
+irb> person = Person.new(1)
 irb> person.to_model == person
 => true
+```
+
+If your model does not act like an Active Model object, then you should
+define`:to_model` yourself returning a proxy object that wraps your object with
+Active Model compliant methods.
+
+```ruby
+class Person
+  def to_model
+    # a proxy object that wraps your object with Active Model compliant methods.
+    PersonModel.new(self)
+  end
+end
+```
+
+#### to_key
+
+The `to_key` method returns an array of the object's key attributes if any of
+the attributes are set, whether or not the object is persisted. Returns nil if
+there are no key attributes.
+
+```irb
 irb> person.to_key
-=> nil
+=> [1]
+```
+
+NOTE: A key attribute is an attribute that is used to identify the object. For
+example, in a database-backed model, the key attribute is the primary key.
+
+#### to_param
+
+The `to_param` method returns a `string` representation of the object's key
+suitable for use in URLs, or `nil` in the case where `persisted?` is `false`.
+
+```irb
 irb> person.to_param
```

Comment:
Both the `to_key` and `to_param` return `nil` when I try this example...hm.
```
>> person=Person.new(1)
=> #<Person:0x00007fe0c7c97688 @id=1>
>> person.to_model
=> #<Person:0x00007fe0c7c97688 @id=1>
>> person.to_model == person
=> true
>> person.to_key
=> nil
>> person.to_param
=> nil
>> person.to_partial_path
=> "people/person"
```

Reply:
Eek, I forgot to update the example with the `attribute_accessor`, thank you!